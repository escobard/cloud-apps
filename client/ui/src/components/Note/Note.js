import React from "react";
import PropTypes from "prop-types";

import { Grid } from "semantic-ui-react";

import "./Note.scss";

// TODO - improve grid structure for error cases
const Note = ({ data: { subject, note, date, icon } }) => (
  <Grid
    columns="equal"
    className={subject === "API error" ? "error note" : "note"}
  >
    <Grid.Column mobile={3} tablet={2} computer={1} className="icon">
      <i
        aria-label={icon && icon.label ? icon.label : "note icon"}
        className={`${(icon && icon.image) ||
          "sticky note"} big circular icon error`}
      />
    </Grid.Column>
    <Grid.Column mobile={9} tablet={12} computer={13} className="content">
      <h3>{subject}</h3>
      <p>{note}</p>
    </Grid.Column>
    {date && (
      <Grid.Column mobile={4} tablet={2} computer={2} className="date">
        <p>{date}</p>
      </Grid.Column>
    )}
  </Grid>
);

Note.propTypes = {
  data: PropTypes.shape({
    subject: PropTypes.string.isRequired,
    note: PropTypes.string.isRequired,
    date: PropTypes.string,
    icon: PropTypes.shape({
      image: PropTypes.string,
      label: PropTypes.string
    })
  })
};

Note.defaultProps = {
  data: {
    date: undefined,
    icon: undefined
  }
};

export default Note;
