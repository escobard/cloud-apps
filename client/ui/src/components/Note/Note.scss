@import "../../styles/variables";

// really useful way of finding the elements that contain some id characters - should consider this approach with rebank
.note {
  border-bottom: $border-gray;
  margin: 0 !important;
  &.error {
    &:hover{
      cursor: initial;
    }
    .column {
      &.icon {
        i {
          color: $red;
          border: solid 2px $red;
        }
      }
    }
  }
  &:hover{
    cursor: pointer;
  }
  .column{
    padding-left: 0 !important;
    &.icon{
      padding-right: 0 !important;
      min-width: 40px;
      i{
        color: $blue;
        border: solid 2px $blue;
        box-shadow: none !important;
        font-size: $font-size-xxl !important;

      }
    }
    &.date{
      text-align: right;
      float: right;
      padding-right: 0 !important;
      p{
        width: 60px;
        float: right;
      }
    }
    &.content{
      padding-right: 0 !important;
      p{
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }
    p{
      color: $gray-medium-dark;
      font-weight: 400;
      word-break: break-word;
    }
  }
}
