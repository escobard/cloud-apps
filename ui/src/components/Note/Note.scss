@import "../../styles/variables";

// really useful way of finding the elements that contain some id characters - should consider this approach with rebank
div[id*="-note"] {
  border-bottom: $border-gray;
  margin: 0 !important;
  padding-bottom: $space-med !important;
  i{
    color: $blue;
    border: solid 2px $gray-light;
    box-shadow: none !important;
    font-size: $font-size-xxxl !important;
  }
  .column{
   padding-left: 0 !important;
    &.date{
      text-align: right;
      padding-right: 0 !important;
    }
    &.content{
      padding-left: $space-lrg !important;
    }
    p{
      color: $gray-medium-dark;
      font-weight: 400;
    }
  }
}
