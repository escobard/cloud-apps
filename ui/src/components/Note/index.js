import React from "react"
import { Grid } from "semantic-ui-react";

const Note = ({ id, data: { title, description, date }}) => {
  return(
    <Grid id={`${id}-note`}  columns='equal'>
      <Grid.Column>
        <i aria-hidden="true" className="sticky note big circular icon"></i>
      </Grid.Column>
      <Grid.Column computer mobile={10} tablet={12} width={14}>
        <h3>{title}</h3>
        <p>{description}</p>
      </Grid.Column>
      <Grid.Column>
        <p>{date}</p>
      </Grid.Column>
    </Grid>
  )
}

export default Note;
