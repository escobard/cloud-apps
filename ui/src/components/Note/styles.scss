@import "../../styles/variables";

// really useful way of finding the elements that contain some id characters - should consider this approach with rebank
div[id*="-note"] {
  border-bottom: $border-gray;
  margin: 0 !important;
  padding-bottom: $space-med !important;
  i{
    color: $blue;
  }
  .column{
   padding-left: 0 !important;
    &.date{
      text-align: right;
      padding-right: 0 !important;
    }
    p{
      color: $gray-medium-dark;
      font-weight: 400;
    }
  }
}
