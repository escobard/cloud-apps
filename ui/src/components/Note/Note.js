import React from "react";
import { Grid } from "semantic-ui-react";

import "./Note.scss";

/** Note Component
 * @name Note
 * @param {string} id, inherited ID from parent
 * @param {object} data, note subject, date and message properties
 * @return {Component} Note
 * */

const Note = ({ id, data: { subject, note, date } }) => (
  <Grid id={`${id}-note`} columns="equal">
    <Grid.Column>
      <i aria-hidden="true" className="sticky note big circular icon" />
    </Grid.Column>
    <Grid.Column mobile={9} tablet={12} computer={14} className="content">
      <h3>{subject}</h3>
      <p>{note}</p>
    </Grid.Column>
    <Grid.Column className="date">
      <p>{date}</p>
    </Grid.Column>
  </Grid>
);

export default Note;
