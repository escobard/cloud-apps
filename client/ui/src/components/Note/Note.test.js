import Note from "./index";

const props = {
  id: "test",
  data: {
    subject: "Test subject",
    note: "Test note",
    date: "02/23/20, 11:46 PM"
  }
};

describe("Note", () => {
  afterAll(() => {
    cleanup();
  });

  it(">> snapshot is up to date", () => {
    const { container } = render(<Note {...props} />);
    expect(container).toMatchSnapshot();
  });

  it(">> should display notes data and circular icon", () => {
    const {
      data: { subject, note, date }
    } = props;
    const { getByText, container } = render(<Note {...props} />);
    expect(getByText(subject));
    expect(getByText(note));
    expect(getByText(date));
    expect(container.querySelector(".sticky.note"));
  });

  it(">> should hide date when no date is given", () => {
    props.data.date = undefined;
    const { container } = render(<Note {...props} />);
    expect(container.querySelector(".date")).toEqual(null);
  });

  it(">> should show different icon if icon is defined", () => {
    props.data.icon = {
      image: "some icon",
      label: "No Notes"
    };
    const { container } = render(<Note {...props} />);
    expect(container.querySelector(".sticky.note")).toEqual(null);
    expect(container.querySelector(".some.icon"));
  });
});
