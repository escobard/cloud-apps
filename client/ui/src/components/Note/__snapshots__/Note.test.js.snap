// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note >> snapshot is up to date 1`] = `
<div>
  <div
    class="ui equal width grid note"
  >
    <div
      class="one wide computer three wide mobile two wide tablet column icon"
    >
      <i
        aria-label="note icon"
        class="sticky note big circular icon error"
      />
    </div>
    <div
      class="thirteen wide computer nine wide mobile twelve wide tablet column content"
    >
      <h3>
        Test subject
      </h3>
      <p>
        Test note
      </p>
    </div>
    <div
      class="two wide computer four wide mobile two wide tablet column date"
    >
      <p>
        02/23/20, 11:46 PM
      </p>
    </div>
  </div>
</div>
`;
